File: frontend/src/app/models/Subject.ts
import {MarksStorage} from "./MarksStorage";
import {Professor} from "./Professor";

export class Subject{

    constructor(
        public id: number,
        public name: string,
        public description: string,
        public marksStorages: MarksStorage[],
        public professors: Professor[]
    ) {
    }
}

File: backend/src/main/java/ru/mondayish/models/Subject.java
package ru.mondayish.models;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Data;

import javax.persistence.*;
import java.util.List;

@Entity
@Table(name = "subjects")
@Data
public class Subject {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;
    private String name;
    private String description;

    @OneToMany(mappedBy = "subject")
    @JsonIgnore
    private List<MarksStorage> marksStorages;

    @ManyToMany(mappedBy = "subjects")
    @JsonIgnore
    private List<Professor> professors;


}
